desc/objectives/_omnigenity.py

```diff
@@ -540,6 +541,348 @@ def compute(self, params, constants=None):
+        errorif(eq_grid.sym, msg="eq_grid must not be symmetric")
+        errorif(field_grid.sym, msg="field_grid must not be symmetric")
+        errorif(eq_grid.num_rho != 1, msg="eq_grid must be a single surface")
+        errorif(field_grid.num_rho != 1, msg="field_grid must be a single surface")
+        errorif(
+            eq_grid.nodes[eq_grid.unique_rho_idx, 0]
+            != field_grid.nodes[field_grid.unique_rho_idx, 0],
+            msg="eq_grid and field_grid must be the same surface",
+        )
+
+        timer = Timer()
+        if verbose > 0:
+            print("Precomputing transforms")
+        timer.start("Precomputing transforms")
+
+        profiles = get_profiles(self._eq_data_keys, obj=eq, grid=eq_grid)
+        eq_transforms = get_transforms(
+            self._eq_data_keys,
+            obj=eq,
+            grid=eq_grid,
+            M_booz=M_booz,
+            N_booz=N_booz,
+        )
+        field_transforms = get_transforms(
+            self._field_data_keys,
+            obj=field,
+            grid=field_grid,
+        )
+
+        # compute returns points on the grid of the field (dim_f = field_grid.num_nodes)
+        # so set quad_weights to the field grid
+        # to avoid it being incorrectly set in the super build
+        w = field_grid.weights
+        w *= jnp.sqrt(field_grid.num_nodes)
+
+        self._constants = {
+            "eq_profiles": profiles,
+            "eq_transforms": eq_transforms,
+            "field_transforms": field_transforms,
+            "quad_weights": w,
+            "helicity": self.helicity,
+        }
+
+        if self._eq_fixed:
+            # precompute the eq data since it is fixed during the optimization
+            eq_data = compute_fun(
+                "desc.equilibrium.equilibrium.Equilibrium",
+                self._eq_data_keys,
+                params=self._eq.params_dict,
+                transforms=self._constants["eq_transforms"],
+                profiles=self._constants["eq_profiles"],
+            )
+            self._constants["eq_data"] = eq_data
+        if self._field_fixed:
+            # precompute the field data since it is fixed during the optimization
+            field_data = compute_fun(
+                "desc.magnetic_fields.OmnigenousField",
+                self._field_data_keys,
+                params=self._field.params_dict,
+                transforms=self._constants["field_transforms"],
+                profiles={},
+                helicity=self._constants["helicity"],
+            )
+            self._constants["field_data"] = field_data
+
+        timer.stop("Precomputing transforms")
+        if verbose > 1:
+            timer.disp("Precomputing transforms")
+
+        if self._normalize:
+            # average |B| on axis
+            self._normalization = jnp.mean(field.B_lm[: field.M_B])
+            errorif(
+                self._normalization <= 0,
+                "|B| on axis is not positive! Check B_lm input.",
```

Comment:
this is avg |B|, the avg could be positive and an incorrect input with negative values somewhere could still be supplied right? maybe adding an errorif earlier in the build that checks if any of the monotnoic spline knot values are negative?